redisinsight/ui/src/pages/vector-search/saved-queries/SavedQueriesScreen.spec.tsx

```diff
@@ -0,0 +1,187 @@
+    )
+  })
+
+  it('should call onQueryInsert with correct query value for second button', () => {
+    render(<SavedQueriesScreen {...defaultProps} />)
+
+    const insertButtons = screen.getAllByText('► Insert')
+
+    // Click second insert button
+    fireEvent.click(insertButtons[1])
+    expect(mockOnQueryInsert).toHaveBeenCalledWith(
+      'FT.SEARCH idx:bike "@type:road"',
+    )
+  })
+
+  it('should render field tags for the selected index', () => {
+    render(<SavedQueriesScreen {...defaultProps} />)
+
+    // The tags should be rendered
+    defaultProps.selectedIndex.tags
+      .map((tag) => tag.toUpperCase())
+      .forEach((tag) => {
+        expect(screen.getByText(tag)).toBeInTheDocument()
+      })
+  })
+
+  describe('with different selected index', () => {
+    it('should render queries for restaurant index when selected', () => {
+      const propsWithRestaurantIndex = {
+        ...defaultProps,
+        selectedIndex: mockSavedIndexes[1], // restaurant_index
+      }
+
+      render(<SavedQueriesScreen {...propsWithRestaurantIndex} />)
+
+      expect(screen.getByText('Search for restaurants')).toBeInTheDocument()
+
+      const insertButtons = screen.getAllByText('► Insert')
+      expect(insertButtons).toHaveLength(1) // 1 query in restaurant index
+    })
+
+    it('should call onQueryInsert with restaurant query values', () => {
+      const propsWithRestaurantIndex = {
+        ...defaultProps,
+        selectedIndex: mockSavedIndexes[1], // restaurant_index
+      }
+
+      render(<SavedQueriesScreen {...propsWithRestaurantIndex} />)
+
+      const insertButtons = screen.getAllByText('► Insert')
+
+      fireEvent.click(insertButtons[0])
+      expect(mockOnQueryInsert).toHaveBeenCalledWith(
+        'FT.SEARCH idx:restaurant "@cuisine:Italian"',
+      )
+    })
+  })
+
+  describe('with empty queries', () => {
+    it('should handle index with no queries', () => {
+      const indexWithNoQueries: SavedIndex = {
+        value: 'empty_index',
+        tags: ['text'],
+        queries: [],
+      }
+
+      const propsWithEmptyQueries = {
+        ...defaultProps,
+        savedIndexes: [indexWithNoQueries],
+        selectedIndex: indexWithNoQueries,
+      }
+
+      render(<SavedQueriesScreen {...propsWithEmptyQueries} />)
+
+      expect(screen.queryByText('► Insert')).not.toBeInTheDocument()
+    })
+  })
+
+  describe('Component Structure', () => {
+    it('should render the main wrapper components', () => {
+      const { container } = render(<SavedQueriesScreen {...defaultProps} />)
+
+      expect(container.firstChild).toBeInTheDocument()
+    })
+
+    it('should have proper button text for insert actions', () => {
+      render(<SavedQueriesScreen {...defaultProps} />)
+
+      const insertButtons = screen.getAllByText('► Insert')
+      insertButtons.forEach((button) => {
+        expect(button).toBeInTheDocument()
+        expect(button.tagName).toBe('BUTTON')
```

Comment:
Why do we care about the type of element? I think we shouldn't test implementation details, I don't see the value of such tests.